clippy_lints/src/methods/mod.rs

```diff
@@ -1515,6 +1556,9 @@ impl<'tcx> LateLintPass<'tcx> for Methods {
             ["unwrap_or_else", ..] => unnecessary_lazy_eval::lint(cx, expr, arg_lists[0], "unwrap_or"),
             ["get_or_insert_with", ..] => unnecessary_lazy_eval::lint(cx, expr, arg_lists[0], "get_or_insert"),
             ["ok_or_else", ..] => unnecessary_lazy_eval::lint(cx, expr, arg_lists[0], "ok_or"),
+            ["or_insert_with", ..] => {
+                lint_entry_or_insert_with_default(cx, method_spans[0].with_hi(expr.span.hi()), arg_lists[0])
```

Comment:
I don't think you should need to do span splicing

Reply:
Can you elaborate? I confess I didn't quite understand this part and just tried to adapt what was being done for the others.

Reply:
Actually, never mind, some of the others do span splicing too. I just don't think the splicing should be _here_, but if you're merging this with the `check_unwrap_or_default` thing then it's going to go away from here anyway.